.github/workflows/ghcr-image-build-and-publish.yml

```diff
@@ -31,7 +31,7 @@ jobs:
 
     steps:
       - name: Checkout repository
-        uses: actions/checkout@v4.2.2
+        uses: actions/checkout@v4
```

Comment:
OpenSSF, CNCF, etc. rather suggest us to pin the specific tag with the commit hash
- https://github.com/lima-vm/lima/pull/2738

Reply:
I don't think this makes practical sense for GitHub owned actions (though it would for third party actions).

Then again, very likely none of our dependencies are safe, even against third-party.

Even containerd itself half-assed the job:
https://github.com/containerd/containerd/blob/main/.github/actions/install-go/action.yml#L13

One way or the other, this changeset here does not impact security or change that situation (4.2.2 is the same amount of risk as 4).

Happy to change them all to sha if this is desirable - though this can come as a different PR.

Let me know your thoughts.

Reply:
Ok, let's get this moving.
Using sha everywhere then.

PTAL at your convenience.